### PR TITLE
perlPackages.HTTPTinyCache: init at 0.002

### PR DESCRIPTION
**Pull request by stigtsp**
###### Motivation for this change

Adds the `HTTP::Tiny::Cache` Perl module

dependencies:
perlPackages.FileUtilTempdir: init at 0.034
perlPackages.Logger: init at 0.037

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>perl530Packages.FileUtilTempdir</li>
    <li>perl530Packages.HTTPTinyCache</li>
    <li>perl530Packages.Logger</li>
    <li>perl532Packages.FileUtilTempdir</li>
    <li>perl532Packages.HTTPTinyCache</li>
    <li>perl532Packages.Logger</li>
  </ul>
</details>

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 106291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>perl530Packages.FileUtilTempdir</li>
    <li>perl530Packages.HTTPTinyCache</li>
    <li>perl530Packages.Logger</li>
    <li>perl532Packages.FileUtilTempdir</li>
    <li>perl532Packages.HTTPTinyCache</li>
    <li>perl532Packages.Logger</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106291` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>perl530Packages.FileUtilTempdir</li>
    <li>perl530Packages.HTTPTinyCache</li>
    <li>perl530Packages.Logger</li>
    <li>perl532Packages.FileUtilTempdir</li>
    <li>perl532Packages.HTTPTinyCache</li>
    <li>perl532Packages.Logger</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>perl530Packages.FileUtilTempdir</li>
    <li>perl530Packages.HTTPTinyCache</li>
    <li>perl530Packages.Logger</li>
    <li>perl532Packages.FileUtilTempdir</li>
    <li>perl532Packages.HTTPTinyCache</li>
    <li>perl532Packages.Logger</li>
  </ul>
</details>

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -8266,6 +8266,22 @@ let
     };
   };
 
+  FileUtilTempdir = buildPerlPackage {
+    pname = "File-Util-Tempdir";
+    version = "0.034";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/File-Util-Tempdir-0.034.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:8273]:
```suggestion
    version = "0.034";
    src = fetchurl rec {
      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/File-Util-Tempdir-${version}.tar.gz";
```

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -9884,6 +9900,22 @@ let
      };
   };
 
+  HTTPTinyCache = buildPerlPackage {
+    pname = "HTTP-Tiny-Cache";
+    version = "0.002";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/HTTP-Tiny-Cache-0.002.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:9907]:
```suggestion
  HTTPTinyCache = buildPerlPackage red {
    pname = "HTTP-Tiny-Cache";
    version = "0.002";
    src = fetchurl {
      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/HTTP-Tiny-Cache-${version}.tar.gz";
```

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -11531,6 +11563,21 @@ let
     buildInputs = [ PathTiny TestWarn ];
   };
 
+  Logger = buildPerlPackage {
+    pname = "Log-ger";
+    version = "0.037";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/Log-ger-0.037.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:11570]:
```suggestion
  Logger = buildPerlPackage rec {
    pname = "Log-ger";
    version = "0.037";
    src = fetchurl {
      url = "mirror://cpan/authors/id/P/PE/PERLANCAR/Log-ger-${version}.tar.gz";
```

**Comment by stigtsp** [comment]
@SuperSandro2000 Thanks for the reviews! 

However, I'd like to avoid adding  `${version}` to `src.url`: It means makes automatic updating having to handle interpolate variables, or more likely means adding two special cases to parsing, something which I'd like to avoid if possible. (It's also a pattern only used by only 10 derivations i `perl-packages.nix`)

PS: I've started working on a new update script after volth left (and cpan2nix)  -> https://github.com/stigtsp/nix-update-perl-packages. It's still in early phase, but thinking to model after `update-python-libraries.py`. Contributions welcome.

**Comment by divi255** [comment]
Good day,

you can get cpan2nix at https://gitee.com/volth/cpan2nix/

volth has a problem with his github account at the moment

**Pull request merged**